ansible_base/authentication/authenticator_plugins/ldap.py

```diff
@@ -528,3 +592,15 @@ def get_or_build_user(self, username, ldap_user):
         )
 
         return user, created
+
+
+def find_class_in_modules(class_name: str) -> object:
+    """
+    Used to find ldap subclasses by string
+    """
+    module_search_space = [config, sys.modules[__name__]]
+    for m in module_search_space:
+        cls = getattr(m, class_name, None)
+        if cls:
+            return cls
+    return None
```

Comment:
Could we make this simpler?

```suggestion
def get_group_class(class_name: str) -> object:
    """
    Used to find ldap subclasses by string
    """
    if class_name == 'PosixUIDGroupType':
        return PosixUIDGroupType
    return getattr(config, class_name, None)
```